src/custom/state/fee/hooks.ts

```diff
@@ -0,0 +1,31 @@
+import { useDispatch, useSelector } from 'react-redux'
+
+import { AppDispatch, AppState } from 'state'
+import { updateFee, clearFee } from './actions'
+import { FeeInformationState, FeeInformation } from './reducer'
+
+interface AddFeeParams extends ClearFeeParams {
+  fee: FeeInformation
+}
+interface ClearFeeParams {
+  token: string // token address
+}
+
+type AddFeeCallback = (addTokenParams: AddFeeParams) => void
+type ClearFeeCallback = (clearTokenParams: ClearFeeParams) => void
+
+export const useFee = (tokenAddress?: string): FeeInformation | undefined => {
+  const { feesMap } = useSelector<AppState, FeeInformationState>(state => state.fee)
+
+  return tokenAddress ? feesMap[tokenAddress]?.fee : undefined
```

Comment:
Put the whole thing inside `useSelector`
react-redux does clever strict equals in store listener and rerenders only if needed
Currently will rerender for you if anything in `feesMap` changes,doesn't have to

Reply:
cool thanks